src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -16917,6 +16917,11 @@ void Unit::_EnterVehicle(Vehicle* vehicle, int8 seatId, AuraApplication const* a
         }
     }
 
+    // For vehicles, check if there is unit type non move
+    // If so, root the vehicle once someone gets in
+    if (HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_DISABLE_MOVE))
```

Comment:
I'd have to check if it works without that. But, I'm fairly sure that when I removed any of these changes there was some situation where I could make the rooted vehicle move.